crates/fuel-core/src/service/adapters/consensus_parameters_provider.rs

```diff
@@ -135,16 +136,15 @@ impl RunnableTask for Task {
                         }
                         Err(err) => {
                             tracing::error!("Failed to cache consensus parameters: {:?}", err);
-                            should_continue = false;
-                            return Ok(should_continue)
+                            return TaskRunResult::Stop
                         }
                     }
                 }
                 should_continue = true;
             }
         }
 
-        Ok(should_continue)
+        TaskRunResult::should_continue(should_continue)
```

Comment:
I though the idea was to remove `should_continue` variable and replace with the usage of the enum=D

Reply:
My goal was to change the interface. 

I'm happy to do some more refactoring too.